File: Python/cinManagement.py
cin_info = None

def manage_cin():
    global cin_info

    while True:
        sub_menu_2 = int(input("Option de gestion du CIN (1. Enregistrement, 2. Mise à jour, 3. Consultation, 4.Retour ) : "))

        if sub_menu_2 not in [1, 2, 3, 4]:
            print("Veuillez saisir une option valable")
            continue

        if sub_menu_2 == 1:
            if cin_info is None:
                print("Enregistrement des informations du CIN...")

                numero_cin = input("Numéro du CIN : ")
                nom_cin = input("Nom sur le CIN : ")
                date_naissance_cin = input("Date de naissance sur le CIN : ")

                if numero_cin.strip() == "" or nom_cin.strip() == "" or date_naissance_cin.strip() == "":
                    print("Veuillez remplir tous les champs du CIN.")
                else:
                    cin_info = {
                        "numero": numero_cin,
                        "nom": nom_cin,
                        "date_naissance": date_naissance_cin,
                    }

                    print("Informations du CIN enregistrées avec succès.")
            else:
                print("Des informations de CIN existent déjà. Vous ne pouvez pas enregistrer de nouveau CIN.")
        elif sub_menu_2 == 2:
            if cin_info is not None:
                print("Option de mise à jour et de renouvellement du CIN...")

                # Affichez les informations actuelles du CIN
                print("Informations actuelles du CIN :")
                print("Numéro du CIN : " + cin_info["numero"])
                print("Nom sur le CIN : " + cin_info["nom"])
                print("Date de naissance sur le CIN : " + cin_info["date_naissance"])

                # Collectez les nouvelles informations du CIN
                nouveau_numero_cin = input("Nouveau numéro du CIN (Appuyez sur Entrée pour conserver l'ancien numéro) : ") or cin_info["numero"]
                nouveau_nom_cin = input("Nouveau nom sur le CIN (Appuyez sur Entrée pour conserver l'ancien nom) : ") or cin_info["nom"]
                nouvelle_date_naissance_cin = input("Nouvelle date de naissance sur le CIN (Appuyez sur Entrée pour conserver l'ancienne date) : ") or cin_info["date_naissance"]

                # Vérifiez que les champs ne sont pas vides
                if nouveau_numero_cin.strip() == "" or nouveau_nom_cin.strip() == "" or nouvelle_date_naissance_cin.strip() == "":
                    print("Veuillez remplir tous les champs du CIN.")
                else:
                    # Mettez à jour les informations du CIN
                    cin_info["numero"] = nouveau_numero_cin
                    cin_info["nom"] = nouveau_nom_cin
                    cin_info["date_naissance"] = nouvelle_date_naissance_cin
                    # Mettez à jour d'autres champs si nécessaire

                    print("Informations du CIN mises à jour avec succès.")
            else:
                print("Aucune information de CIN existante. Veuillez enregistrer des informations de CIN d'abord.")
        elif sub_menu_2 == 3:
            if cin_info is not None:
                print("Consultation des informations du CIN...")

                print("Informations actuelles du CIN :")
                print("Numéro du CIN : " + cin_info["numero"])
                print("Nom sur le CIN : " + cin_info["nom"])
                print("Date de naissance sur le CIN : " + cin_info["date_naissance"])
            else:
                print("Aucune information de CIN existante. Veuillez enregistrer des informations de CIN d'abord.")
        elif sub_menu_2 == 4:
            print("Retour au menu principal.")
            break

File: JS/cinManagement.js
const prompt = require("prompt-sync")();
let cinInfo = null; 

function manageCIN() {
    while (true) {
        let subMenu2 = +prompt("Option de gestion du CIN (1. Enregistrement, 2. Mise à jour, 3. Consultation, 4.Retour ) : ");

        if (![1, 2, 3, 4, 5].includes(subMenu2)) {
            console.log("Veuillez saisir une option valable");
            continue;
        }
        if (subMenu2 === 1) {
            if (cinInfo == null) {
                console.log("Enregistrement des informations du CIN...");
        
                let numeroCIN = prompt("Numéro du CIN : ");
                let nomCIN = prompt("Nom sur le CIN : ");
                let dateNaissanceCIN = prompt("Date de naissance sur le CIN : ");
        
                if (numeroCIN.trim() === "" || nomCIN.trim() === "" || dateNaissanceCIN.trim() === "") {
                    console.log("Veuillez remplir tous les champs du CIN.");
                } else {
                    cinInfo = {
                        numero: numeroCIN,
                        nom: nomCIN,
                        dateNaissance: dateNaissanceCIN,
                    };
        
                    console.log("Informations du CIN enregistrées avec succès.");
                }
            } else {
                console.log("Des informations de CIN existent déjà. Vous ne pouvez pas enregistrer de nouveau CIN.");
            }
        }              
        else if (subMenu2 === 2) {
            if (cinInfo != null) {
                console.log("Option de mise à jour et de renouvellement du CIN...");
        
                // Affichez les informations actuelles du CIN
                console.log("Informations actuelles du CIN :");
                console.log("Numéro du CIN : " + cinInfo.numero);
                console.log("Nom sur le CIN : " + cinInfo.nom);
                console.log("Date de naissance sur le CIN : " + cinInfo.dateNaissance);
        
                // Collectez les nouvelles informations du CIN
                let nouveauNumeroCIN = prompt("Nouveau numéro du CIN (Appuyez sur Entrée pour conserver l'ancien numéro) : ") || cinInfo.numero;
                let nouveauNomCIN = prompt("Nouveau nom sur le CIN (Appuyez sur Entrée pour conserver l'ancien nom) : ") || cinInfo.nom;
                let nouvelleDateNaissanceCIN = prompt("Nouvelle date de naissance sur le CIN (Appuyez sur Entrée pour conserver l'ancienne date) : ") || cinInfo.dateNaissance;
        
                // Vérifiez que les champs ne sont pas vides
                if (nouveauNumeroCIN.trim() === "" || nouveauNomCIN.trim() === "" || nouvelleDateNaissanceCIN.trim() === "") {
                    console.log("Veuillez remplir tous les champs du CIN.");
                } else {
                    // Mettez à jour les informations du CIN
                    cinInfo.numero = nouveauNumeroCIN;
                    cinInfo.nom = nouveauNomCIN;
                    cinInfo.dateNaissance = nouvelleDateNaissanceCIN;
                    // Mettez à jour d'autres champs si nécessaire
        
                    console.log("Informations du CIN mises à jour avec succès.");
                }
            } else {
                console.log("Aucune information de CIN existante. Veuillez enregistrer des informations de CIN d'abord.");
            }
        } 
        else if (subMenu2 === 3) {
            if (cinInfo != null) {
                console.log("Consultation des informations du CIN...");
        
                console.log("Informations actuelles du CIN :");
                console.log("Numéro du CIN : " + cinInfo.numero);
                console.log("Nom sur le CIN : " + cinInfo.nom);
                console.log("Date de naissance sur le CIN : " + cinInfo.dateNaissance);
            } else {
                console.log("Aucune information de CIN existante. Veuillez enregistrer des informations de CIN d'abord.");
            }
        }        
        else if (subMenu2 === 4) {
            console.log("Retour au menu principal.");
            break;
        }
    }
}

module.exports = {
    manageCIN
};
